synadm/cli/notice.py

```diff
@@ -0,0 +1,92 @@
+#
+# synadm is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# synadm is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""server notice-related CLI commands
+"""
+
+import re
+import click
+from click_option_group import optgroup, MutuallyExclusiveOptionGroup
+from click_option_group import RequiredAnyOptionGroup
+
+from synadm import cli
+
+@cli.root.group()
+def notice():
+    """ Send server notices to local users
+    """
+
+@notice.command(name="send")
+@click.option("--from-file/--from-argument", "-f/-a", default=False,
+    help="""Interpret arguments as file paths instead of notice content and 
+    read content from those files""")
+@click.option("--paginate", type=int, default=100,
+    help="Sets how many users will be retrieved from the server at once")
+@click.argument("to", type=str, default=None)
+@click.argument("plain", type=str, default=None)
+@click.argument("formatted", type=str, default=None, required=False)
+@click.pass_obj
+def notice_send_cmd(helper, from_file, paginate, to, plain, formatted):
+    """Send server notices to local users.
+
+    TO - either a matrix ID (e.g. '@abc:example.com') or a regular expression 
+        as used in python (e.g. '^.*' to send to all users)
+    
+    PLAIN - plain text content of the notice
+    
+    FORMATTED - (HTML-) formatted content of the notice. If not set, PLAIN 
+        will be used.
+    """
+    if from_file:
+        with open(plain, "r") as plain_file:
+            plain_content = plain_file.read()
+        if formatted:
+            with open(formatted, "r") as formatted_file:
+                formatted_content = formatted_file.read()
+        else:
+            formatted_content = plain_content
+    else:
+        plain_content = plain
+        formatted_content = formatted
+    
+    def confirm_prompt(users):
+            if helper.batch:
+                return True
+            prompt = "Recipients (list may be incomplete):\n"
+            ctr = 0
+            for user in users:
+                if not re.search(to, user) is None:
+                    prompt = prompt + " - " + user + "\n"
+                    ctr = ctr + 1
+                    if ctr >= 10:
+                        prompt = prompt + " - ..."
+                        break
+            prompt = prompt + "\nUnformatted message:\n---\n" + plain_content\
+                + "\n---\nFormatted message:\n---\n" + formatted_content\
+                + "\n---\nSend now?"
```

Comment:
A possible solution for above mentioned problem might be to, instead of concatenating strings with `+`, making use of f-strings (`f'Plaintext {variable}')` or a strings `format()` method. I think both would handle the type None by translating it it to the string "None", not sure if that would be the perfect solution here or the problem needs to be solved somewhere else.

Reply:
Oh you found it yourself. Adding the extra edge case if-branch is necessary anyways because the formatted content variable must be set to send the message (this notice module will never send plain messages, if the user supplies only plain text it send a formatted message with plain content for both text variants). So that is the more robust solution I guess.